packages/cactus-plugin-ccmodel-hephaestus/src/main/typescript/plugin-ccmodel-hephaestus.ts

```diff
@@ -0,0 +1,911 @@
+  }
+
+  get numberEventsLog(): number {
+    return this.crossChainLog.numberEvents();
+  }
+
+  get numberEventsUnmodeledLog(): number {
+    return this.unmodeledEventLog.numberEvents();
+  }
+
+  public purgeNonConformedEvents(): void {
+    this.nonConformedCrossChainLog.purgeLogs();
+  }
+
+  get numberEventsNonConformedLog(): number {
+    return this.nonConformedCrossChainLog.numberEvents();
+  }
+
+  public purgeCrossChainEvents(): void {
+    this.crossChainLog.purgeLogs();
+  }
+
+  public getInstanceId(): string {
+    return this.instanceId;
+  }
+
+  public getCaseId(): string {
+    return this.caseID;
+  }
+
+  public setIsModeling(bool: boolean): void {
+    this.isModeling = bool;
+  }
+
+  public setCaseId(id: string): void {
+    this.unmodeledEventLog.purgeLogs();
+    this.caseID = id;
+  }
+
+  public async onPluginInit(): Promise<unknown> {
+    return;
+  }
+
+  public async shutdown(): Promise<void> {
+    this.log.info(`Shutting down...`);
+    const serverMaybe = this.getHttpServer();
+    if (serverMaybe.isPresent()) {
+      this.log.info(`Awaiting server.close() ...`);
+      const server = serverMaybe.get();
+      await promisify(server.close.bind(server))();
+      this.log.info(`server.close() OK`);
+    } else {
+      this.log.info(`No HTTP server found, skipping...`);
+    }
+  }
+
+  async registerWebServices(app: Express): Promise<IWebServiceEndpoint[]> {
+    const webServices = await this.getOrCreateWebServices();
+    await Promise.all(webServices.map((ws) => ws.registerExpress(app)));
+    return webServices;
+  }
+
+  public async getOrCreateWebServices(): Promise<IWebServiceEndpoint[]> {
+    if (Array.isArray(this.endpoints)) {
+      return this.endpoints;
+    }
+
+    const { log } = this;
+
+    log.info(`Installing web services for plugin ${this.getPackageName()}...`);
+
+    const endpoints: IWebServiceEndpoint[] = [];
+
+    // TODO implement endpoints
+
+    const pkg = this.getPackageName();
+    log.info(`Installed web services for plugin ${pkg} OK`, { endpoints });
+
+    return endpoints;
+  }
+
+  public getHttpServer(): Optional<Server | SecureServer> {
+    return Optional.ofNullable(this.httpServer);
+  }
+
+  public getPackageName(): string {
+    return `@hyperledger/cactus-plugin-ccmodel-hephaestus`;
+  }
+
+  private createReceiptFromRunTransactionV1ExchangeBesu(
+    data: RunTransactionV1ExchangeBesu,
+    caseId: string,
+  ): BesuV2TxReceipt {
+    return {
+      caseID: caseId,
+      blockchainID: LedgerType.Besu2X,
+      timestamp: data.timestamp,
+      transactionID: data.response.transactionReceipt.transactionHash,
+      from: data.response.transactionReceipt.from,
+      invocationType: data.request.invocationType,
+      methodName: data.request.methodName,
+      parameters: data.request.params,
+      gasUsed: data.response.transactionReceipt.gasUsed,
+      gasPrice: data.request.gasPrice as number,
+    };
+  }
+
+  private pollTxReceiptsBesu(
```

Comment:
@RafaelAPB Does this do any actual polling? It looks like it just massages the  data. If I'm understanding it correctly then I'd recommend renaming it to something expressing that instead of polling for tx receipts.

Reply:
@brunoffmateus Great, thank you!